### PR TITLE
Add pre/postinstall scripts to prevent 'npm install' failure

### PR DESCRIPTION
**Pull request by jmgaya**
## **TL;DR;**

Just fix the `npm install` script

## Explanation

The recommended Node Version for this repository is `10.15.1`, which uses npm `6.4.1` under the hood (see `.nvmrc`).

When trying to install dependencies directly from Git repositories, this version of NPM uses internally the [git: protocol](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols#The-Git-Protocol), which performs operations such as:

* `/usr/bin/git ls-remote -h -t git://github.com/<organization>/<repository>.git`
* `git clone --depth=1 -q -b master git://github.com/<organization>/<repository>.git <some-cache-directory>`

However, Github dropped support for this protocol [a while ago](https://github.blog/2021-09-01-improving-git-protocol-security-github/), hence `npm install` fails always.

The less elegant, but most pragmatic solution @Josmorsot and I found during the journey is to redirect every `git:` protocol request through `https:` by modifying the NPM configuration locally

### CODE REVIEW

**Review by Josmorsot — APPROVED** [review]
🚀

**Review by VictorVelarde — CHANGES_REQUESTED** [review]
I know this looks harmless, but let's talk before moving on

**Comment by VictorVelarde** [comment]
If this goes against cloud-native branch, just for our do-catalog bundle, I'm fine with it.
